packages/connect/src/interceptor.spec.ts

```diff
@@ -154,36 +156,44 @@ describe("stream interceptors", () => {
   ] as const;
   it("promise client", async () => {
     const log: string[] = [];
-    const transport = stubTransport({
-      interceptors: [
-        makeLoggingInterceptor("outer", log),
-        makeLoggingInterceptor("inner", log),
-      ],
-      streamResponseHeader: new Headers({
-        "stream-response-header": "foo",
-      }),
-      streamResponseTrailer: new Headers({
-        "stream-response-trailer": "foo",
-      }),
-    });
 
-    const input = createAsyncIterable([
-      new TestService.methods.serverStream.I(),
-    ]);
+    const transport = createRouterTransport(
+      ({ service }) => {
+        service(TestService, {
+          serverStream: (request: Int32Value, context: HandlerContext) => {
+            context.responseHeader.set("stream-response-header", "response-header");
+            context.responseTrailer.set("stream-response-trailer", "response-trailer");
+            return createAsyncIterable([{ value: request.value.toString() }]);
+          },
+        });
+      },
+      {
+        transport: {
+          interceptors: [
+            makeLoggingInterceptor("outer", log),
+            makeLoggingInterceptor("inner", log),
+          ],
+        },
+      }
+    );
 
-    const res = await transport.stream(
+    const response = await transport.stream(
       TestService,
       TestService.methods.serverStream,
       undefined,
       undefined,
       {
-        "stream-request-header": "foo",
+        "stream-request-header": "request-header",
       },
-      input
+      createAsyncIterable([new Int32Value({ value: 42 })])
     );
-    for await (const m of res.message) {
-      expect(m).toEqual(new StringValue({ value: "" }));
+
+    for await (const message of response.message) {
+      expect(message).toEqual(new StringValue({ value: "42" }));
     }
+
     expect(log).toEqual(wantLog);
+    expect(response.header.get("stream-response-header")).toEqual("response-header");
+    expect(response.trailer.get("stream-response-trailer")).toEqual("response-trailer");
```

Comment:
sounds good.  I only included them because the prior/existing test was already testing headers (with `wantLog`) and this just adds _also_ checking the values.  I also agree on the second part: if a better place arises to situate this stuff it'd be great to move it in the future.

thanks!